client/my-sites/plan-features-2023-grid/billing-timeframe.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+	isWpComFreePlan,
+	isWpcomEnterpriseGridPlan,
+	PLAN_BIENNIAL_PERIOD,
+	PLAN_ANNUAL_PERIOD,
+} from '@automattic/calypso-products';
+import { getCurrencyObject } from '@automattic/format-currency';
+import { localize, TranslateResult, useTranslate } from 'i18n-calypso';
+import { FunctionComponent } from 'react';
+import { useSelector } from 'react-redux';
+import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
+
+interface Props {
+	planName: string;
+	billingTimeframe: TranslateResult;
+	billingPeriod: number | null | undefined;
+	rawPrice: number | null;
+	maybeDiscountedFullTermPrice: number | null;
+	annualPricePerMonth: number | null;
```

Comment:
These optional nulls are kinda unfortunate, stemming from how the underlying selectors/structures behave. I think the alternative, here, would be to condition where the component is imported/used, and instead set these to just the expected types (`number`). Not sure if worth/better though.